File: service/app/src/test/java/io/miragon/blueprint/application/service/CancelInsurancePolicyServiceTest.java
package io.miragon.blueprint.application.service;

import static io.miragon.blueprint.domain.leasing.TestObjectBuilder.testLeasingApplication;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.verifyNoMoreInteractions;

import io.miragon.blueprint.application.port.outbound.InsurancePort;
import io.miragon.blueprint.domain.leasing.LeasingApplication;
import org.junit.jupiter.api.Test;

class CancelInsurancePolicyServiceTest {

    private final InsurancePort insurance = mock(InsurancePort.class);
    private final CancelInsurancePolicyService underTest = new CancelInsurancePolicyService(insurance);

    @Test
    void cancelPolicyDelegatesTheCompensationToTheInsurer() {
        // given: an application whose policy must be revoked
        LeasingApplication application = testLeasingApplication().build();
        // when: the policy is cancelled
        underTest.cancelPolicy(application.id());
        // then: the insurance out-port revokes the policy
        verify(insurance).cancelPolicy(application.id());
        verifyNoMoreInteractions(insurance);
    }
}

File: service/app/src/test/kotlin/io/miragon/blueprint/application/service/CancelInsurancePolicyServiceTest.kt
package io.miragon.blueprint.application.service

import io.miragon.blueprint.application.port.outbound.InsurancePort
import io.miragon.blueprint.domain.leasing.testLeasingApplication
import io.mockk.Runs
import io.mockk.confirmVerified
import io.mockk.every
import io.mockk.just
import io.mockk.mockk
import io.mockk.verify
import org.junit.jupiter.api.Test

class CancelInsurancePolicyServiceTest {

    private val insurance = mockk<InsurancePort>()
    private val underTest = CancelInsurancePolicyService(insurance = insurance)

    @Test
    fun `cancelPolicy delegates the compensation to the insurer`() {
        // given: an application whose policy must be revoked
        val application = testLeasingApplication()
        every { insurance.cancelPolicy(application.id) } just Runs
        // when: the policy is cancelled
        underTest.cancelPolicy(application.id)
        // then: the insurance out-port revokes the policy
        verify { insurance.cancelPolicy(application.id) }
        confirmVerified(insurance)
    }
}
